pkg/service/restore/service.go

```diff
@@ -0,0 +1,280 @@
+	}
+	if scyllaClient == nil {
+		return nil, errors.New("invalid scylla provider")
+	}
+
+	return &Service{
+		repairSvc:      repairSvc,
+		session:        session,
+		config:         config,
+		metrics:        metrics,
+		scyllaClient:   scyllaClient,
+		clusterSession: clusterSession,
+		logger:         logger,
+	}, nil
+}
+
+// Restore creates and initializes worker performing restore with given properties.
+func (s *Service) Restore(ctx context.Context, clusterID, taskID, runID uuid.UUID, properties json.RawMessage) error {
+	s.logger.Info(ctx, "Restore",
+		"cluster_id", clusterID,
+		"task_id", taskID,
+		"run_id", runID,
+	)
+
+	w, err := s.newWorker(ctx, clusterID)
+	if err != nil {
+		return errors.Wrap(err, "create worker")
+	}
+	w.setRunInfo(taskID, runID)
+
+	if err := w.initTarget(ctx, properties); err != nil {
+		return errors.Wrap(err, "init target")
+	}
+
+	if w.target.Continue {
+		if err := w.decorateWithPrevRun(ctx); err != nil {
+			return err
+		}
+	}
+
+	if w.run.PrevID == uuid.Nil {
+		// Reset metrics on fresh start
+		w.metrics.ResetClusterMetrics(w.run.ClusterID)
+		// Cache must be initialised only once, as it contains information already lost
+		// in the cluster (e.g. tombstone_gc mode, views definition, etc).
+		if err := w.initUnits(ctx); err != nil {
+			return errors.Wrap(err, "initialize units")
+		}
+		if err := w.initViews(ctx); err != nil {
+			return errors.Wrap(err, "initialize views")
+		}
+	} else {
+		w.clonePrevProgress(ctx)
+		// Check that all units are still present after resume
+		for _, u := range w.run.Units {
+			for _, t := range u.Tables {
+				ok, err := w.client.TableExists(ctx, "", u.Keyspace, t.Table)
+				if err != nil {
+					return errors.Wrapf(err, "query table %s.%s existence", u.Keyspace, t.Table)
+				}
+				if !ok {
+					return errors.Wrapf(err, "table %s.%s doesn't exist", u.Keyspace, t.Table)
+				}
+			}
+		}
+	}
+	w.insertRun(ctx)
+
+	if w.target.RestoreTables {
+		var totalBytesToRestore int64
+		for _, unit := range w.run.Units {
+			totalBytesToRestore += unit.Size
+		}
+		tw := newTablesWorker(w, s.repairSvc, totalBytesToRestore)
+		err = tw.restore(ctx)
+	} else {
+		sw := &schemaWorker{worker: w}
+		err = sw.restore(ctx)
+	}
+
+	if err == nil {
+		w.run.Stage = StageDone
+		w.insertRun(ctx)
+	}
+	return err
+}
+
+// GetTarget returns validated target from properties.
+func (s *Service) GetTarget(ctx context.Context, clusterID uuid.UUID, properties json.RawMessage) (Target, error) {
+	w, err := s.newWorker(ctx, clusterID)
+	if err != nil {
+		return Target{}, errors.Wrap(err, "create worker")
+	}
+	defer w.clusterSession.Close()
+
+	if err := w.initTarget(ctx, properties); err != nil {
+		return Target{}, errors.Wrap(err, "initialize target")
+	}
+	return w.target, nil
+}
+
+// GetUnits returns validated units from properties.
+func (s *Service) GetUnits(ctx context.Context, clusterID uuid.UUID, properties json.RawMessage) ([]Unit, error) {
```

Comment:
This method is not used.
The only references I found comes from integration tests.

```
TestRestoreGetUnitsIntegration
TestRestoreGetUnitsErrorIntegration
```

Please remove if it's unused.
Please update listed integration tests to use GetTargetUnitsViews instead of GetUnits